Clarification for HTTP/2 support with Kestrel on Windows8.1/Server2012R2 (#9522)
aspnet/Docs: aspnetcore/fundamentals/servers/index.md

  * Public-facing edge server connections use HTTP/2, but the reverse proxy connection to Kestrel uses HTTP/1.1.
  * Target framework: Not applicable to IIS out-of-process deployments.

> > [!NOTE]

Author: Is there a better way to add footnotes in markdown files? Ideally, it should be an asterisk in the section above that links to this note.

Reviewer: This looks good to me

Reviewer 2: Use `&dagger;` (single dagger) and `&Dagger;` (double dagger). I'm trying to sneak more daggers 🔪 into the docs.

> > [!NOTE]

Reviewer 3: You should update the kestrel doc as well.
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel?view=aspnetcore-2.2#http/2-support

> > Kestrel has limited support for HTTP/2 on Windows Server 2012 R2/Windows 8.1 because the list of supported TLS cipher suites available on these versions of the operating system is limited. A certificate generated using an Elliptic Curve Digital Signature Algorithm (ECDSA) may be required to ensure a compatible cipher suite can be successfully negotiated during TLS handshake.

Reviewer: Don't we want to mention this is only a problem in Chrome/firefox?

Author: It depends on the cipher suites advertised by the client during TLS handshake. I didn't want to make this clarification specific to particular browsers but I also wanted to cover the case for the most popular ones that may run into this problem.

Reviewer 2: > during TLS handshake

Is it **_only_** during the handshake? ... I understand that the supported ciphers are compared for the selection of which one will be used during the handshake, but it seems like the entire passage should be worded to indicate that an ECDSA-based cert (algorithm) might be required to secure the entire connection.

Thinking along the lines of ...

```suggestion
> Kestrel has limited support for HTTP/2 on Windows Server 2012 R2 and Windows 8.1 because the list of supported TLS cipher suites available on these operating systems is limited. A certificate generated using an Elliptic Curve Digital Signature Algorithm (ECDSA) may be required to secure TLS connections.
```

Author: When I tested without the ECDSA certs, the negotiation with Chrome/Firefox ended with a cipher suite that doesn't satisfy the minimum security requirements for HTTP/2. They have a blacklist of ciphers https://tools.ietf.org/html/rfc7540#appendix-A that are not secure enough.

Reviewer 3: Wait, I thought the EC cert was all you needed to make it work?
